### PR TITLE
338 pre commit hooks

### PR DESCRIPTION
**Pull request by Jday7879**
# Summary

Add your summary here - keep it brief, to the point, and in plain English.

# Checklists

This pull request meets the following requirements:

- [x] installable with all dependencies recorded
- [x] runs without error
- [x] follows PEP8 and project specific conventions
- [ ] appropriate use of comments, for example no descriptive comments
- [ ] functions documented using Numpy style docstings
- [ ] assumptions and decisions log considered and updated if appropriate
- [ ] unit tests have been updated to cover essential functionality for a reasonable range of inputs and conditions
- [ ] other forms of testing such as end-to-end and user-interface testing have been considered and updated as required
- [x] tests suite passes (locally as a minimum)
- [ ] peer reviewed with review recorded

If you feel some of these conditions do not apply for this pull request, please
add a comment to explain why.

This branch has been created to perform the bulk changes required by the pre-commit hooks. 
No functions have been added as a result of this branch. 
One manual change to code has been made in `flag_and _count_matched_pairs.py` on L112

### CODE REVIEW

**Review by AntonZogk — APPROVED** [review]
All looking good with pre-commit hooks, was also able to install the requirements and the checks run after commit. Just 2 minor comments. 

This pull request has changes to src/ and test/ files, had a closer look at them and all were related to code formatting from the hooks (removing whitespace, end liner etc) 

The requirements.txt contains typically dependencies for user to install/run  the source code while in this pull request it looks like that it has dependencies for developers and contributors, it is worth having this separately (in the toml or setup.cfg file), we can have though a separate ticket for this in the future.

**Comment by Jday7879** [comment]
Thanks @AntonZogk for the review. 
I think we have a ticket on the backlog to separate out requirements into users and development (and probably docs).
I'll add your review comments onto that ticket for future reference as well
